Bring architecture doc up to current thinking. (#12)
AMWA-TV/ms-05-01: docs/Device Model.md

- When a blockspec is intended to be reused, the blockspec ID value must be unique within the intended universe of use.  This unique value is called a Global Spec ID (GSID) value - see [Global blockspec IDs](#global-blockspec-ids), below.

A blockspec is not a block, it is only the description of a block. Blockspecs are the standardized way of documenting the contents of NCA blocks.

A blockspec is a static block definition.  NCA also supports blocks (_dynamic blocks_) whose contents are modifiable at runtime.  See [Device configurability](#device-configurability).

Within a device, a blockspec may be instantiated any number of times.  For example, one might create a blockspec that defines an audio mixer channel, then instantiate it 64 times to define a 64-channel mixer.

Blockspec relationships are illustrated in the figure [Blockspec](#blockspec), below.

<span id="blockspec"></span>

| ![The ncBlock abstract class, a block class, two blocks, and related documents](images/Blockspec.svg) |
|:--:|
| *Blockspec:  The ncBlock abstract class, a block class, two blocks, and related documents* |

> On the left is the control model with its WebIDL definition files. In this case, only two control model classes are shown: **ncObject** , which is the parent of everything, and `ncBlock` , which is the abstract base class upon which all block classes are built.

Reviewer: ```suggestion
On the left of the figure is the control model with its WebIDL definition files. In this case, only two control model classes are shown: **ncObject** , which is the parent of everything, and `ncBlock` , which is the abstract base class upon which all block classes are built.
```

## Blocks

> The control model defines a class named **ncaBlock** that is a container for objects. Subclasses of **ncaBlock** are called **block classes**. Each block class defines a specific configuration of objects.

Reviewer: ```suggestion
The control model defines a class named **ncaBlock** that is a container for objects. In Devices, block classes are instantiated to make objects known as **blocks**.
```

> In Devices, block classes are instantiated to make objects known as **blocks.** A block class may be instantiated as many times as required to define the required control API. For example, an eight-channel audio mixer might define a block class named **myMixChannel** and instantiate it eight times, to create eight blocks which define the eight required channel-control APIs.

Reviewer: ```suggestion
```

### Nested blocks

> A block may contain other blocks. The contained blocks are said to be **nested** inside the containing block. For example, the **myMixChannel** class might specify an instance of a block named **EqA** that defines an equalizer section. **EqA** is "nested" inside **myMixChannel**.

Reviewer: ```suggestion
A block may contain other blocks. The contained blocks are said to be **nested** inside the containing block.
```

### Block namespaces

Every NCA object has a name that is unique _within its containing block_ (see [Identification](Identification.md), and every block constitutes a namespace for the objects it contains.  This name is known as the _**role name**_.

> If a block class is instantiated multiple times in a device, the role names of its contained objects (e.g. `Gain`, `Mute`, `Pad`,...) will not conflict, because each of the eight block objects is a separate namespace.

Reviewer: ```suggestion
```

Every NCA object is uniquely identified by its _**role path**_.  See [**Identification(Role path)**](Identification.md#RolePath).

> <a name="Blockspecs"/> ## Blockspecs and Blockspec IDs ##

Reviewer: ```suggestion
## Blockspecs and Blockspec IDs
```
